### DISCUSSION TITLE
Organização do frontend

### DISCUSSION THREAD

**Discussion by solilorys** [frontendbr/forum#2293] [Dicas]
Olá pessoal, tenho uma dúvida sobre a organização do código do frontend. Estou fazendo um projeto pessoal, mas não sei muito bem como organizar o código de uma maneira rápida, fácil e prática para quem observa. Eu fiz algumas coisas como, colocar os h1, h2, h3 um separado do outro e ir codando com os códigos do CSS de acordo com a ordem no HTML. Precisa colocar os h1, h2, h3, juntos ou seguido? 

Se alguém tiver dicas de como organizar um documento HTML, CSS e JS pode me passar dicas, aulas, sites, qualquer coisa para me ajudar nessa jornada de conhecimento! Obrigada.

**Comment by felquis** [comment]
Dicas:

Não existe uma regra bem definida e obrigatória, então você pode escolher o que bem entender que funciona para você sem medo. Mesmo com tudo bem organizado, quando alguém novo aparecer, essa pessoa vai precisar de um tempo para entender o projeto completamente. É impossível alguém entender um código sem de fato estuda-lo.

Num geral uma dica que dou é organizar tudo como uma livraria, dessa forma se você encontra uma sessão, você pelo menos pode saber que ali estão todos os livros da categoria ou pelo menos quase.

Algo que me ajuda muito, é ter um README.md que explica como instalar dependências, como rodar em localhost isso já ajuda muito a quebrar o gelo. Num geral essas docs de como fazer coisas ficam desatualizadas com o tempo, mas faz parte de empresas grande ficar continuamente atualizando essas docs de acordo com a realidade que se transforma.

O item acima de aplica caso use algum framework, ou linguagem de programação, no readme já coloca links para as documentações oficiais das mesmas que as pessoas podem usar como referências para entender o código que você fez usando-as.

Aguardo as dicas de outros internautas rsrs

**Reply by felquis** [reply]
As perguntas que você fez são bem focadas no front-end que você tem, neste caso se você postar exatamente o código que fez poderemos opinar de forma mais cirúrgica direto onde você está desconfiando que pode melhorar.

Mas lembre-se que opinião alheia deve ser filtrada e se vc está entendendo bem o código como está hoje, pode ser melhor do que mudar para algo que você precisará pensar muito para entender diminuindo seu domínio sobre o próprio código.

**Reply by solilorys** [reply]
Entendi! To terminando o projeto ainda, deveria postar como está agora (em andamento) ou quando terminar e alguém dar uma revisada legal?

**Reply by felquis** [reply]
O povo gosta de opinar então em andamento ou pronto, terá ideias em ambas as fases.

**Comment by nazarepiedady** [comment]
Minha cara colega de profissão, eu recomendo que acompanhar o canal do @tarciozemel,
ele produz vídeos muito úteis sobre ferramentas e técnicas de organização de código front-end,
além do mais ele já fez vídeos explicando a aplicabilidade de metodologias como BEM, SMACSS e
OOCSS.

O seu sítio oficial onde ele faz muita questão de trazer conteúdo relevante sobre desenvolvimento para
werb é este: [desenvolvimentoparaweb](https://desenvolvimentoparaweb.com)

A ligação para o seu canal é esta: [canal dpw](https://www.youtube.com/channel/UCdHcHgSrWidiOg-mNFNB1Nw)

**Reply by solilorys** [reply]
Obrigada!

**Comment by ghost** [comment]
A melhor coisa a se fazer é: separe seu código por responsabilidades. No HTML é possível deixar o CSS tudo dentro de uma tag `<style>` e deixar todo seu CSS lá, e essa tag vai dentro da tag `<head>`. Caso use scripts, poderia fazer o mesmo: deixe o conteúdo todo dentro da tag `<script>` e deixe essa tag logo antes de fechar a tag `</body>`.

Com isso vocë terá uma estrutura parecida com:

```
html
'- head
  '- style
'- body
  '- [seu conteúdo HTML]
  '- script
```

Só evite colocar estilos e scripts diretamente dentro de tags HTML utilizando [classes](https://developer.mozilla.org/pt-BR/docs/Web/HTML/Global_attributes/class).

Você também pode delegar estilos e scripts num arquivo separado, aí você fica com um arquivo contendo somente código HTML, outro contendo somente código css e um terceiro só com código de scripts.

Para organizar seu código HTML, o ideal seria pensar em semântica. [Semântica em HTML](https://developer.mozilla.org/pt-BR/docs/Glossary/Semantics) seria utilizar tags de acordo com seu significado. Sobre estruturação de conteúdo de documento, pense que uma página tem uma estrutura igual a de um livro. Ao escrever estruturadamente, não é uma boa prática colocar títulos de peso menor se não for pertencente a um título maior. Pense sempre em títulos delimitando conteúdos.

```
h1 (condiz ao título da página)
  h2 - Assunto 1 (um agrupamento maior de conteúdo mas não necessariamente o total)
    h3 - Assunto 1.1 (Um sub assunto do Assunto 1)
    h3 - Assunto 1.2 (Um sub assunto do Assunto 1)
  h2 - Assunto 2 (um agrupamento maior de conteúdo mas não necessariamente o total)
  ...
```

...e por aí vai. Algo bem parecido com escrever um Livro, os títulos devem refletir algo como se fosse o sumário. E com isso você atinge um conteúdo mais taxonômico para quem lê. Ideal que ao tirar os estilos, você consiga ler o conteúdo da página de uma forma lógica e isso refletindo no teu código.

Só com isso você já atinge uma certa legibilidade de código.

Isso lhe ajudou em algo? Ou não era bem isso? Se entendi errado a questão, peço desculpas. Ou se quiser que eu aprofunde em algum quesito...

**Reply by solilorys** [reply]
Muito obg!
